### PR TITLE
[OSRS] Fixes Incorrect filler item names causing failures on tests.

### PR DESCRIPTION
**Pull request by digiholic**
## What is this fixing or adding?
Fixes issue reported by Silviris on the world-dev discord channel. The `get_filler_item_name` function had some obsolete item names in it, causing generation failures. Names have been updated to match what they currently are in the CSV data package.

## How was this tested?
Ran the attached weights YAML from the discord (cannot attach here due to unsupported file type) with at least 5 players. Previously always caused errors, now it does not.

### CODE REVIEW

File: worlds/osrs/__init__.py [hunk 1]
```diff
@@ -524,7 +524,10 @@ def create_region(self, name: str) -> "Region":
         return region
 
     def create_item(self, item_name: str) -> "Item":
-        item = [item for item in item_rows if item.name == item_name][0]
+        items = [item for item in item_rows if item.name == item_name]
+        if len(items) == 0:
```
Comment (Exempt-Medic) [inline worlds/osrs/__init__.py:528]: Do you expect to actually hit this in AP releases? You could use an assert instead

File: worlds/osrs/__init__.py [hunk 2]
```diff
@@ -524,7 +524,10 @@ def create_region(self, name: str) -> "Region":
...
+            raise f"No matching item found for name {item_name} for player {self.player_name}"
```
Comment (Exempt-Medic) [inline worlds/osrs/__init__.py:529]: This should probably be an Exception and not just `raise`, right?

File: worlds/osrs/__init__.py (hunk 1 above)
Comment (digiholic) [reply worlds/osrs/__init__.py:528]: I had asserts in other spots before but was suggested to raise an exception instead. I'm not entirely sure the difference, is this a situation where assert would be better?

Comment (Exempt-Medic) [reply worlds/osrs/__init__.py:528]: An assert will not happen in released versions of AP. Only in source/dev versions

Comment (digiholic) [reply worlds/osrs/__init__.py]: Fair enough. This wouldn't be a runtime exception so it's been replaced with an assert

**Review by Exempt-Medic — APPROVED** [review]
Change LGTM

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**